includes/classes/SiteAutomation/Integration.php

```diff
@@ -52,31 +68,49 @@ public function get_curated_posts( $posts, $query ) {
 		}
 
 		$curated_response = $this->request->get( $query_vars['sophi_curated_page'], $query_vars['sophi_curated_widget'] );
+		$request_status   = $this->request->get_status();
 
-		// Determine how we should format the results  based on the fields parameter.
-		$fields = $query->get( 'fields', '' );
+		if ( $request_status['success'] ) {
```

Comment:
As it seems we are not guaranteeing that `$this->request->get_status()` will return an array nor if that array will contain a `'success'` index, it would be to check if  `! empty( $request_status['success'] )`. What do you think @oscarssanchez? Otherwise, I think people can get a `not defined` warning.